submit/controllers/ui/new/upload_delete.py

```diff
@@ -0,0 +1,251 @@
+"""
+Controllers for file-delete-related requests.
```

Comment:
I think this will also be needed for replacements, so could potentially live one level up.

Reply:
I was thinking that a bunch of these would need to go one level up due to replacements.

Reply:
I'll make an issue for this.

Reply:
https://arxiv-org.atlassian.net/browse/ARXIVNG-2950